onnxruntime/core/providers/cuda/nn/conv.cc

```diff
@@ -173,37 +236,52 @@ Status Conv<T>::ComputeInternal(OpKernelContext* context) const {
     }
 
     if (!y_data) {
-      Tensor* Y = context->Output(0, TensorShape(s_.y_dims));
+      Y = context->Output(0, TensorShape(s_.y_dims));
       // special case when there is a dim value of 0 in the shape.
       if (Y->Shape().Size() == 0)
         return Status::OK();
 
-      y_data = reinterpret_cast<CudaT*>(Y->template MutableData<T>());
+      if (!s_.post_slicing_required) {
+        y_data = reinterpret_cast<CudaT*>(Y->template MutableData<T>());
+      } else {
+        // Post slicing needed. Create and fill in the Conv results in an intermediate buffer.
+        IAllocatorUniquePtr<void> temp = GetScratchBuffer<void>(TensorShape(s_.y_dims_with_adjusted_pads).Size() * element_size);
+        memory_for_cudnn_conv_results = std::move(temp);
+        y_data = reinterpret_cast<CudaT*>(memory_for_cudnn_conv_results.get());
+      }
     }
 
     const auto alpha = Consts<CudaT>::One;
     const auto beta = Consts<CudaT>::Zero;
 
     IAllocatorUniquePtr<void> workspace = GetScratchBuffer<void>(s_.workspace_bytes);
 
-    CUDNN_RETURN_IF_ERROR(cudnnConvolutionForward(CudnnHandle(),
-                                                  &alpha,
-                                                  s_.x_tensor,
-                                                  x_data,
-                                                  s_.filter_desc,
-                                                  w_data,
-                                                  s_.conv_desc,
-                                                  s_.algo,
-                                                  workspace.get(),
-                                                  s_.workspace_bytes,
-                                                  &beta,
-                                                  s_.y_tensor,
-                                                  y_data));
+    (cudnnConvolutionForward(CudnnHandle(),
+                             &alpha,
```

Comment:
Accidental delete of CUDNN_RETURN_IF_ERROR?

Reply:
Fixed